win32ss/reactx/ntddraw/dxeng.c

```diff
@@ -465,7 +465,11 @@ DxEngGetDCState(HDC hDC,
                 break;
             case 2:
                 /* Return the number of rectangles in the visible region. */
-                retVal = (DWORD_PTR) pDC->prgnRao ? pDC->prgnRao->rdh.nCount : pDC->prgnVis->rdh.nCount;
+                /* It should not be 1, because dxg!DxDdReenableDirectDrawObject
+                 * does not accept 1. So in case it's equal to 1, don't use it.
+                 * 2 is what typically returned in Windows Server 2003 SP2.
+                 */
+                retVal = (DWORD_PTR) pDC->prgnVis->rdh.nCount != 1 ? pDC->prgnVis->rdh.nCount : 2;
```

Comment:
This is not a number of rectangles, but a region complexity.
Try using `REGION_Complexity(pDC->prgnVis);` instead. This will also do the extra NULL check on prgnVis that was missing in your original code (and return an adequate value).
Or `pDC->prgnVis->prgnattr->iComplexity`, but I'm not sure whether this field is always in sync with what `REGION_Complexity` could return.

Reply:
Thanks @HBelusca. Committed the fix. :+1: Calling `REGION_Complexity()` works for me. And it also returns `2`, so this is indeed more correct fix. That's why it returns `2` in Windows. It's logically obvious. How I didn't understood that earlier... :slightly_smiling_face:

Reply:
I'm also not sure that `pDC->prgnVis->prgnattr->iComplexity` is updated correctly when it is passed to the function.